flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssigner.java

```diff
@@ -158,10 +159,10 @@ private MySqlSnapshotSplitAssigner(
 
     @Override
     public void open() {
-        chunkSplitter.open();
         discoveryCaptureTables();
         captureNewlyAddedTables();
         startAsynchronouslySplit();
+        chunkSplitter.open();
```

Comment:
`startAsynchronouslySplit` should be called after `chunkSplitter` initialized ? 
Current logic we build `chunkSplitter` in constructor and `captureNewlyAddedTables` is a little confused.